cmd/crank/beta/trace/internal/resource/xrm/client.go

```diff
@@ -113,31 +116,56 @@ func getResourceChildrenRefs(r *resource.Resource, getConnectionSecrets bool) []
 	if xrcNamespace := obj.GetNamespace(); xrcNamespace != "" {
 		// This is an XRC, get the XR ref, we leave the connection secret
 		// handling to the XR
-		xrc := claim.Unstructured{Unstructured: obj}
-		if ref := xrc.GetResourceReference(); ref != nil {
-			refs = append(refs, v1.ObjectReference{
-				APIVersion: ref.APIVersion,
-				Kind:       ref.Kind,
-				Name:       ref.Name,
-			})
-		}
-		if getConnectionSecrets {
-			xrcSecretRef := xrc.GetWriteConnectionSecretToReference()
-			if xrcSecretRef != nil {
-				ref := v1.ObjectReference{
-					APIVersion: "v1",
-					Kind:       "Secret",
-					Name:       xrcSecretRef.Name,
-					Namespace:  xrcNamespace,
+		out := &v1.ObjectReference{}
+		// Determine if this is a namespaced XR or an XRC
+		// XRCs will have the spec.crossplane.resourceRef or spec.resourceRef field
+		// If either field is found, we treat it as an XRC
+		checkNewField := fieldpath.Pave(obj.Object).GetValueInto("spec.crossplane.resourceRef", out)
+		checkLegacyField := fieldpath.Pave(obj.Object).GetValueInto("spec.resourceRef", out)
+		if checkNewField == nil || checkLegacyField == nil {
+			xrc := claim.Unstructured{Unstructured: obj}
+			if ref := xrc.GetResourceReference(); ref != nil {
+				refs = append(refs, v1.ObjectReference{
+					APIVersion: ref.APIVersion,
+					Kind:       ref.Kind,
+					Name:       ref.Name,
+				})
+			}
+			if getConnectionSecrets {
+				xrcSecretRef := xrc.GetWriteConnectionSecretToReference()
+				if xrcSecretRef != nil {
+					ref := v1.ObjectReference{
+						APIVersion: "v1",
+						Kind:       "Secret",
+						Name:       xrcSecretRef.Name,
+						Namespace:  xrcNamespace,
+					}
+					refs = append(refs, ref)
 				}
-				refs = append(refs, ref)
 			}
+			return refs
 		}
-		return refs
 	}
 	// This could be an XR or an MR
 	xr := composite.Unstructured{Unstructured: obj}
+	// The xr.GetResourceReferences() function checks for resourceRefs at spec.crossplane.resourceRefs
+	// For legacy XRs, we need to check at spec.resourceRefs path
+	// We can determine if it's a legacy XR by checking for the presence of the claimRef field
```

Comment:
I think this would only catch legacy XRs if they were derived from a claim, but they can also be manually created.  

Perhaps we should check for the presence of `spec.crossplane` to determine _not legacy_ instead.